### ISSUE TITLE
test_dtm_overflow_warnings failing

### ISSUE THREAD

**Issue by mglisse** [GUDHI/gudhi-devel#543]
> 2/2 Test #40: test_dtm_py_test .................***Failed    2.49 sec
> ============================= test session starts ==============================
> platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
> rootdir: /home/glisse/repos/gudhi/build/2021-11-03-19-18-59_GUDHI_3.5.0rc1
> plugins: typeguard-2.13.0, cov-2.12.1, doctestplus-0.11.0, remotedata-0.3.2, pep8-1.0.6, pylint-0.18.0
> collected 5 items
> 
> ../../python/test/test_dtm.py ....F                                      [100%]
> 
> =================================== FAILURES ===================================
> __________________________ test_dtm_overflow_warnings __________________________
> 
>     def test_dtm_overflow_warnings():
>         pts = numpy.array([[10., 100000000000000000000000000000.], [1000., 100000000000000000000000000.]])
>         impl_warn = ["keops", "hnsw"]
>     
>         with warnings.catch_warnings(record=True) as w:
>             for impl in impl_warn:
>                 dtm = DistanceToMeasure(2, q=10000, implementation=impl)
>                 r = dtm.fit_transform(pts)
> >           assert len(w) == 2
> E           assert 1 == 2
> E            +  where 1 = len([<warnings.WarningMessage object at 0x7f12ec24d2e0>])
> 
> ../../python/test/test_dtm.py:100: AssertionError
> =========================== short test summary info ============================
> FAILED ../../python/test/test_dtm.py::test_dtm_overflow_warnings - assert 1 == 2
> ========================= 1 failed, 4 passed in 1.52s ==========================

**Comment by Hind-M** [comment]
Does this happen with the current master ? With which configuration ? Are hnswlib and pykeops there ? (If yes, which versions ?)

**Comment by mglisse** [comment]
Yes, with current master, on linux x86_64 with CMAKE_BUILD_TYPE=Release.

-- Found PythonInterp: /usr/bin/python3 (found version "3.9.7") 
++ Python module cython - Version 0.29.24  found
++ Python module pytest - Version 6.2.5  found
++ Python module matplotlib - Version 3.3.4  found
++ Python module numpy - Version 1.19.5  found
++ Python module scipy - Version 1.7.1  found
++ Python module sphinx - Version 4.2.0  found
++ Python module sklearn - Version 0.23.2  found
++ Python module ot - Version 0.7.0  found
++ Python module pybind11 - Version 2.7.1  found
++ Python module torch - Version 1.10.0+cu113  found
++ Python module pykeops - Version 1.5  found
++ Python module eagerpy - Version 0.30.0  found
++ Python module hnswlib found
++ Python module tensorflow - Version 2.7.0  found
++ Python module sphinx_paramlinks - Version 0.5.1  found
++ Python module python_docs_theme found

(hnswlib is 0.5.2)

**Comment by mglisse** [comment]
It seems to be the "keops" case that does not warn.

**Comment by mglisse** [comment]
`q` doesn't have any effect on knn. But even if I set `p`, it does not help unless I also pass `metric="minkowski"`. Maybe I should change the default from "euclidean" to "minkowski" in KNearestNeighbors...

**Comment by mglisse** [comment]
Curiously, in keops, 1e20f squared gives the largest float (around 3e38) and not infinity :confused:

**Comment by Hind-M** [comment]
Hmm, I see. I think that the overflow doesn't happen with keops using gpu. With my computer, since I don't have one, pykeops switches to cpu and does warn ... Maybe we should just check if cuda is available with torch before processing the thrown warnings in the test.

**Comment by mglisse** [comment]
We could indeed check cuda. Or since it looks like it will be fixed in the next version, we could check the version. Or we could disable the test for now and open an issue to reenable it once the next version of pykeops is out. Your test played its role and let us check that the warning works on various platforms, it is probably ok if we don't test it during a few months.
